api/src/main/java/jakarta/data/Insert.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data;
+
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>The {@code Insert} annotation indicates to dynamic templates or repositories that the annotated method
+ * will perform an insert operation. This method should have a unique parameter whose type can be one of the following:
+ * </p>
+ * <ul>
+ *     <li>The entity to be inserted.</li>
+ *     <li>An {@code Iterable} of entities to be inserted.</li>
+ *     <li>An array of entities to be inserted.</li>
+ * </ul>
+ * <p>The return type of the annotated method should match the type of the parameter. For example, if the method is
+ * annotated with {@code @Insert} and takes a parameter of type {@code Car car}, the return type should be {@code Car}.
+ * Similarly, if the parameter is an {@code Iterable<Car>} or an array of {@code Car}, the return type should be the
+ * corresponding type.
```

Comment:
in case of EntityManager it return the entity self, but not all databases have EntityManager.

Reply:
> in case of EntityManager it return the entity self, but not all databases have EntityManager.

You have it backwards.  Please refer to the [EntityManager.persist](https://jakarta.ee/specifications/persistence/3.1/apidocs/jakarta.persistence/jakarta/persistence/entitymanager#persist(java.lang.Object)) JavaDoc which clearly shows a return type of `void`.  It does not return the entity.  If you want the entity returned, you need to use `save` (which corresponds to EntityManager.merge and is the update-or-insert-if-not-present pattern).  The whole point of offering the insert vs save is to let the user optimize for different patterns.  Forcing an insert to also fetch an updated entity (if it's even possible - it won't be in JPA if the id is generated) would be contrary to that.